dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -1138,57 +1165,51 @@ public <T extends IdentifiableObject> List<T> getByAttributeAndValue( Class<T> t
 
     @Override
     @Transactional( readOnly = true )
-    public <T extends IdentifiableObject> boolean isAttributeValueUnique( Class<T> type,
-        T object, AttributeValue attributeValue )
+    public <T extends IdentifiableObject> boolean isAttributeValueUnique( @Nonnull Class<T> type,
+        @Nonnull T object, @Nonnull AttributeValue attributeValue )
     {
         IdentifiableObjectStore<T> store = getIdentifiableObjectStore( type );
         return store != null && store.isAttributeValueUnique( object, attributeValue );
     }
 
     @Override
     @Transactional( readOnly = true )
-    public <T extends IdentifiableObject> boolean isAttributeValueUnique( Class<T> type,
-        T object, Attribute attribute, String value )
+    public <T extends IdentifiableObject> boolean isAttributeValueUnique( @Nonnull Class<T> type,
+        @Nonnull T object, @Nonnull Attribute attribute, @Nonnull String value )
     {
         IdentifiableObjectStore<T> store = getIdentifiableObjectStore( type );
         return store != null && store.isAttributeValueUnique( object, attribute, value );
     }
 
+    @Nonnull
     @Override
     @Transactional( readOnly = true )
-    public <T extends IdentifiableObject> List<T> getAllByAttributeAndValues( Class<T> type,
-        Attribute attribute, List<String> values )
+    public <T extends IdentifiableObject> List<T> getAllByAttributeAndValues( @Nonnull Class<T> type,
+        @Nonnull Attribute attribute, @Nonnull List<String> values )
     {
         IdentifiableObjectStore<T> store = getIdentifiableObjectStore( type );
         return store != null ? store.getAllByAttributeAndValues( attribute, values ) : List.of();
     }
 
+    @Nonnull
     @Override
     @Transactional( readOnly = true )
     public Map<Class<? extends IdentifiableObject>, IdentifiableObject> getDefaults()
     {
-        Long catId = defaultObjectCache.get( Category.class.getName(),
-            key -> getByName( Category.class, DEFAULT ).getId() );
-        Long cateComboId = defaultObjectCache.get( CategoryCombo.class.getName(),
-            key -> getByName( CategoryCombo.class, DEFAULT ).getId() );
-        Long catOptionId = defaultObjectCache.get( CategoryOption.class.getName(),
-            key -> getByName( CategoryOption.class, DEFAULT ).getId() );
-        Long catOptionComboId = defaultObjectCache.get(
```

Comment:
FYI: In theory this was a potential NPE as `getId()` could be invoked on `null` returned by `getByName`